utils/utils.c

```diff
@@ -278,24 +280,43 @@ void fetchGlfsVolServerFromEnv()
   LOG("mgmt", GB_LOG_INFO, "Block Hosting Volfile Server Set to: %s", gbConf.volServer);
 }
 
-
-int
-initLogging(void)
+static int
+initLogDirAndFiles(char *newLogDir)
 {
   char *logDir = NULL;
+  char *tmpLogDir = NULL;
+  int ret = 0;
+
+
+  /*
+   * The priority of the logdir setting is:
+   * 1, /etc/sysconfig/gluster-blockd config file
+   * 2, "GB_LOGDIR" from the ENV setting
+   * 3, default as GB_LOGDIR_DEF
+   */
+  if (newLogDir) {
+    logDir = newLogDir;
+  } else {
+    logDir = getenv("GB_LOGDIR");
 
+    tmpLogDir = glusterBlockDynConfigGetLogDir();
+    if (tmpLogDir) {
+      logDir = tmpLogDir;
+    }
 
-  logDir = getenv("GB_LOGDIR");
-  if (!logDir) {
-    logDir = GB_LOGDIR;
+    if (!logDir) {
+      logDir = GB_LOGDIR_DEF;
+    }
   }
 
   if (strlen(logDir) > PATH_MAX - GB_MAX_LOGFILENAME) {
     fprintf(stderr, "strlen of logDir Path > PATH_MAX: %s", logDir);
     return EXIT_FAILURE;
+    goto out;
```

Comment:
return and goto ? My be you want to assign  ret = EXIT_FAILURE; and goto out;

Reply:
Yeah, will fix it.
Thanks.